swift/ql/lib/codeql/swift/elements/decl/IterableDeclContext.qll

```diff
@@ -1,4 +1,18 @@
-// generated by codegen/codegen.py, remove this comment if you wish to edit this file
 private import codeql.swift.generated.decl.IterableDeclContext
+private import codeql.swift.elements.decl.NominalTypeDecl
+private import codeql.swift.elements.decl.ExtensionDecl
 
-class IterableDeclContext extends Generated::IterableDeclContext { }
+/**
+ * A nominal type (class, struct, enum or protocol) or extension.
+ */
+class IterableDeclContext extends Generated::IterableDeclContext {
+  /**
+   * Gets the `NominalTypeDecl` corresponding to this `IterableDeclContext`
+   * resolving an extension to the extended type declaration.
+   */
+  NominalTypeDecl resolveExtensions() {
```

Comment:
I'm not sure I like the current name. Could we maybe use a name that's more consistent with the QL style-guide (i.e., `getX`)? Maybe something like `getResolvedTypeDecl`?

Reply:
Yeah, I see what you mean...

Reply:
We also use `resolved` a lot to mean something different, so its looking like `resolveExtensions` was overall a pretty poor name.  I've gone with `getNominalTypeDecl` for now, does that sound OK to you?

Reply:
That sounds okay to me, yeah 👍